sources/repo-update

```diff
@@ -154,11 +133,69 @@ fi
 [ -e "${LOCKFILE}" ] && error_mes "lock"
 /usr/bin/touch "${LOCKFILE}" 2> /dev/null || /usr/bin/true
 
-## work block
-# building
+## Work block
+# Building
 echo -e "${bwhite}[II]${cclose} Building packages"
 cd "${STAGINGDIR}"
-/usr/bin/find -name 'PKGBUILD' -type f -execdir /usr/bin/bash -c "func_build "${PREPAREDIR}" "${ROOTDIR}"" \;
+#Get package list in staging folder
+for PACKAGE in $(/usr/bin/find -name 'PKGBUILD' -type f); do
+    cd "${STAGINGDIR}"
+    #Get package folder
+    cd $(dirname "${PACKAGE}")
+    #Get PKGBUILD
+    source PKGBUILD
+    echo -e "${bwhite}[II] ${bblue}=>${cclose} Building ${bwhite}${pkgname}${cclose}"
+    #Build for repo architecture
+    for ARCH in ${ARCHS}; do
+    	#Build for package architecture
+	for _ARCHS in ${arch[@]}; do
+		# Build if package is any architecture
+		if [[ ${_ARCHS} =~ any ]]; then
+        			echo -e "${bwhite}[III] ${bblue}=>${cclose} Build using default tools${cclose}";
+      				LC_MESSAGES=C /usr/bin/sudo ${TOOLSDEFAULT} -r ${ROOTDIR} ${RECHROOT}
+    		else
+    			# Build if package is a lib32 type
+	 		if [[ ${pkgname} =~ lib32 ]]; then
+	 			# Build if repo architecture is equal the package architecture
+      				if [ ${ARCH} == "i686" ] && [ ${_ARCHS} == "i686" ]; then
```

Comment:
as fart as i can see the package will build only if `${ARCH} == ${_ARCHS}", so I believe it may be checked once at the start of the cycle and continue if they don't match, i.e.:
```
[[ ${ARCH} != ${_ARCHS} ]] && continue
```